### PR TITLE
Loading data from save

### PR DESCRIPTION
**Pull request by Zemurin**

### CODE REVIEW

File: CK2ToEU4/Source/CK2World/World.cpp [hunk 1]
```diff
@@ -108,3 +154,86 @@ bool CK2::World::uncompressSave(const std::string& saveGamePath)
...
+		const auto& holder = title.second->getHolder();
+		if (!holder.first) continue; // don't bother with titles without holders.
+		if (liege.first.empty())
+		{
+			// this is a potential indep.
+			potentialIndeps.insert(std::pair(title.first, title.second));
+			continue;
+		}
+		const auto& liegeTitle = liege.second->getTitle();
+		if (liegeTitle.first == "e_hre")
```
Comment (Idhrendur) [inline CK2ToEU4/Source/CK2World/World.cpp:190]: It'd be nice if this was configurable (thinking of a game of mine where I'd like to break up my Roman empire on conversion)

File: CK2ToEU4/Source/CK2World/Characters/Characters.cpp [hunk 2]
```diff
@@ -12,8 +18,137 @@ CK2::Characters::Characters(std::istream& theStream)
...
-		auto newCharacter = Character(theStream, std::stoi(charID));
-		characters.insert(std::pair(newCharacter.getID(), newCharacter));
-		});
+		auto newCharacter = std::make_shared<Character>(theStream, std::stoi(charID));
+		characters.insert(std::pair(newCharacter->getID(), newCharacter));
+		});	
 	registerRegex("[A-Za-z0-9\\_:.-]+", commonItems::ignoreItem);
 }
+
+void CK2::Characters::linkDynasties(const Dynasties& theDynasties)
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/Characters/Characters.cpp:27]: this is useful but I don't comprehend how it works.

File: CK2ToEU4/Source/CK2World/Characters/Characters.cpp [hunk 3]
```diff
@@ -12,8 +18,137 @@ CK2::Characters::Characters(std::istream& theStream)
...
+				{
+					Log(LogLevel::Warning) << "Base title ID: " << title2Itr->first << " has no definition!";
+				}
+			}
+		}
+	}
+	Log(LogLevel::Info) << "<> " << counterPrim << " primary titles and " << counterBase << " base titles linked.";
+}
+
+void CK2::Characters::linkCapitals(const Provinces& theProvinces)
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/Characters/Characters.cpp:129]: thanx, found some bugs with display for this one.

File: CK2ToEU4/Source/CK2World/Characters/Characters.cpp [hunk 4]
```diff
@@ -12,8 +18,137 @@ CK2::Characters::Characters(std::istream& theStream)
...
+{
+	auto counterCapital = 0;
+	const auto& provinces = theProvinces.getProvinces();
+	for (const auto& character : characters)
+	{
+		if (!character.second->getCapital().first.empty())
+		{
+			auto match = false;
+			for (const auto& province: provinces)
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/Characters/Characters.cpp:138]: unknown

[the remark this answers is not in the document]
Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/Characters/Characters.cpp:138]: fixed though.

File: CK2ToEU4/Source/CK2World/World.cpp (hunk 1 above)
Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/World.cpp:190]: We can break up anything, sure, but without HRE dynamics, which are linked to HRE exclusively (or, well, to any single empire we tell it to be), multiple shattered empires will only devolve into simple independent bordergore.

Well, that's 2 options. Once is to set which empire becomes the "HRE" and inherits HRE dynamics, and the other is what to do with empires in general. I'll get on both of these.

Comment (IohannesIohannium) [reply CK2ToEU4/Source/CK2World/World.cpp:190]: I'd suggest to check for a title inside the generic configuration file:

```
#hremechanics: the title posted here will use EU4 HRE mechanics. Please, write the title code correctly
hremechanics = e_hre
#Other common suggestions: e_byzantium, e_roman_empire
```

Comment (Zemurin) [reply CK2ToEU4/Source/CK2World/World.cpp:190]: this will go into config options, no worry.

**Pull request merged**